v3/lints/cabf_smime_br/mailbox_address_from_san.go

```diff
@@ -0,0 +1,117 @@
+package cabf_smime_br
+
+/*
+ * ZLint Copyright 2024 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import (
+	"github.com/zmap/zcrypto/encoding/asn1"
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zcrypto/x509/pkix"
+	"github.com/zmap/zlint/v3/lint"
+	"github.com/zmap/zlint/v3/util"
+)
+
+// MailboxAddressFromSAN - linter to enforce MAY/SHALL NOT requirements for SMIME certificates
+type MailboxAddressFromSAN struct {
+}
+
+func init() {
+	lint.RegisterLint(&lint.Lint{
+		Name:          "e_mailbox_address_shall_contain_an_rfc822_name",
+		Description:   "All Mailbox Addresses in the subject field or entries of type dirName of this extension SHALL be repeated as rfc822Name or otherName values of type id-on-SmtpUTF8Mailbox in this extension",
+		Citation:      "SMIME BRs: 7.1.4.2.1",
+		Source:        lint.CABFSMIMEBaselineRequirements,
+		EffectiveDate: util.ZeroDate,
```

Comment:
Blocking on this one, the others are fiddly code nits/opinions.

```suggestion
		EffectiveDate: util.CABF_SMIME_BRs_1_0_0_Date,
```

Reply:
Thank you for the review! I have addressed this and it prevented the lint from running against the integration tests so now the expected error set is 0. I'll try to raise to my team that we should be using proper effective dates so it can make future contribution easier.